### PR TITLE
Fix smart count for no results message

### PR DESCRIPTION
**Pull request by thibault-barrat**
## Problem

#10291 introduces translation of the resource name in the no results message. Nevertheless, when there is no results, the smart count used for the translation should be 0.

## Solution

Update smart_count when calling `getResourceLabel` function.

### CODE REVIEW

**Review by slax57 — APPROVED** [review]
Indeed, thanks!

**Review by slax57 — CHANGES_REQUESTED** [review]
Tests appear to be failing though. Can you have a look?

**Review by slax57 — APPROVED** [review]
Nice catch, thanks!

File: packages/ra-core/src/core/useGetResourceLabel.ts [hunk 1]
```diff
@@ -39,11 +39,7 @@ export const useGetResourceLabel = (): GetResourceLabel => {
                           smart_count: count,
                           _: resourceDefinition.options.label,
                       })
-                    : humanize(
-                          count > 1
-                              ? pluralize(resource)
-                              : singularize(resource)
-                      ),
+                    : humanize(inflect(resource, count)),
```
Comment (slax57) [inline packages/ra-core/src/core/useGetResourceLabel.ts:42]: Actually `inflection` is a tool to manage strings for the English language, so this fix works well for English, but it may break other languages (like French).
I'll need to make sure it's OK to change this with the team. Let's put this PR on hold for now. Thank you anyways!

Comment (slax57) [reply packages/ra-core/src/core/useGetResourceLabel.ts:42]: Consensus: applying English rules should be the default, using French gramary rules was a bug.
Users that use another language than English should provide translation keys for their resources.
Besides, this will only change the default value for resources that do not have a translation key, and when the smart count is 0, hence making this a pretty limited change.
TLDR: we're OK to merge!

**Review by slax57 — APPROVED** [review]
